kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/SettingUpKolibri.vue

```diff
@@ -189,28 +192,55 @@
       wizardContext(key) {
         return this.wizardService.state.context[key];
       },
-      provisionDevice() {
-        this.$store.dispatch('clearError');
-        client({
-          url: urls['kolibri:core:deviceprovision'](),
-          method: 'POST',
-          data: this.deviceProvisioningData,
-        })
-          .then(() => {
-            const welcomeDismissalKey = 'DEVICE_WELCOME_MODAL_DISMISSED';
-            const facilityImported = 'FACILITY_IS_IMPORTED';
-            window.sessionStorage.setItem(welcomeDismissalKey, false);
-            window.sessionStorage.setItem(
-              facilityImported,
-              this.wizardContext('isImportedFacility'),
-            );
-
-            Lockr.rm('savedState'); // Clear out saved state machine
-            redirectBrowser();
-          })
-          .catch(e => {
-            this.$store.dispatch('handleApiError', { error: e });
+      async startProvisionDeviceTask() {
+        try {
+          await TaskResource.startTask({
+            type: TaskTypes.PROVISIONDEVICE,
+            ...this.deviceProvisioningData,
           });
+          this.pollProvisionTask();
+        } catch (e) {
+          this.$store.dispatch('handleApiError', { error: e });
+        }
+      },
+      async pollProvisionTask() {
+        try {
+          const tasks = await TaskResource.list({ queue: PROVISION_TASK_QUEUE });
+          const [task] = tasks || [];
+          if (!task) {
+            throw new Error('Device provisioning task not found');
+          }
+          if (task.status === TaskStatuses.COMPLETED) {
+            const facilityId = task.result.facility_id;
```

Comment:
Add this to the `extra_metadata` instead, rather than updating the API endpoint to return result.